File: Codelabs (V2)/Unit1/intreactive_ui/src/main/java/dev/emg/intreactive_ui/MainActivity.java
package dev.emg.intreactive_ui;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

  private TextView countTextView;
  private int mCount = 0;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    countTextView = findViewById(R.id.count_tv);
  }

  public void toastButton(View view) {
    Toast.makeText(this, R.string.toast, Toast.LENGTH_SHORT).show();
  }

  public void countButton(View view) {
    ++mCount;
    if (countTextView != null) {
      countTextView.setText(Integer.toString(mCount));
    }
  }
}

File: Kotlin Fundamentals/Lesson1/app/src/main/java/dev/emg/lesson1/MainActivity.kt
package dev.emg.lesson1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import java.util.Random

class MainActivity : AppCompatActivity() {

  private lateinit var resultText: TextView

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    resultText = findViewById(R.id.result_tv)

    val rollButton: Button = findViewById(R.id.roll_btn)
    rollButton.setOnClickListener { rollDice() }

    val countUpButton: Button = findViewById(R.id.countUp_btn)
    countUpButton.setOnClickListener { countUp() }
  }

  private fun rollDice() {
    val randomInt = Random().nextInt(6) + 1
    resultText.text = randomInt.toString()
    Toast.makeText(this, "button clicked", Toast.LENGTH_SHORT)
        .show()
  }

  private fun countUp() {
    when (val currentValue = resultText.text.toString()) {
      "Hello World!" -> resultText.text = "1"
      "6" -> return
      else -> resultText.text = (currentValue.toInt() + 1).toString()
    }
  }
}
